app/models/graph-controller.ts

```diff
@@ -81,10 +81,20 @@ export class GraphController {
         }
 
         // TODO: ...like
+        let likeVal: ElementValue | undefined = undefined;
+        if (like) {
+            const bridge = this.bridges.getB(like);
+            if (bridge) {
+                likeVal = bridge.core;
+            }
+        }
 
-        return validateEdge(this.plugin, source, destination, undefined);
-    }
+        if (!likeVal) {
+            likeVal = new ElementValue(this.activeEdgeType, this.core.environment);
```

Comment:
In that regard, is it possible to get the type information from the arguments of `validateEdgeHandler`? If so, maybe rename `validateEdge` to `typecheckEdge`, only pass element types to `typecheckEdge` and remove the plugin parameter, then `return typecheckEdge(...) && (plugin.validateEdge ? plugin.validateEdge(...) : true)` in `validateEdgeHandler`.

I think this still does not address your concern, though. If `like` is provided as a previously deleted edge, is there any way to get its type information from the environment? I'm not sure this is currently possible.